File: 3841_password_strength/pramoda/solution.py
class Solution:
    def passwordStrength(self, password: str) -> int:
        filtered = set(password)
        score = 0

        for char in filtered:

            if (char.islower()):
                score += 1
            elif (char.isupper()):
                score += 2
            elif (char.isdigit()):
                score += 3
            else:
                score += 5

        return score


s = Solution()
print(s.passwordStrength(password = "bbB11#"))

File: 3841_password_strength/pramoda/solution.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int passwordStrength(char* password) {
    bool seen[94] = {0};

    size_t size = strlen(password);
    int score = 0;
    for (size_t i = 0; i < size; ++i) {
        int diff = password[i] - '!';

        if (seen[diff]) continue;
        if (password[i] >= 'a' && password[i] <= 'z') score += 1;
        if (password[i] >= 'A' && password[i] <= 'Z') score += 2;
        if (password[i] >= '0' && password[i] <= '9') score += 3;
        switch (password[i]) {
        case '!':
        case '@':
        case '#':
        case '$':
            score += 5;
            break;
        default: break;
        }
        seen[diff] = true;;
    }

    return score;
}

int main() {

    printf("%d\n", passwordStrength("aA1!"));

    return 0;
}